MVVM Module/Source/Feed Presentation/FeedViewModel.swift

```
@@ -20,14 +20,18 @@ final class FeedViewModel {
 
 	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
-	
+    var onFeedError: Observer<String>?
```

`onFeedError: Observer<String> ` could be `onErrorStateChange: Observer<String?>`, so the ViewModel decides when to show/hide errors (instead of the view controller). It's often easier to manage the state when all changes are centralized.

For example, in the `loadFeed()` method, the ViewModel can call  `onErrorStateChange(.none)` to hide the error before loading the feed.